### ISSUE TITLE
Editor: a full image edit options modal is missing

### ISSUE THREAD

**Issue by designsimply** [Automattic/wp-calypso#307]
Users would like to have a way to edit image details similar to the edit flow for galleries. They would like to be able to edit everything you have access to modify in WP Admin.

Here is how it currently looks in WordPress.com:

![screen shot 2015-11-20 at fri nov 20 2 22 32 pm](https://cloud.githubusercontent.com/assets/1119271/11309537/64ce41d2-8f92-11e5-990e-48637167f35b.png)

![screen shot 2015-11-20 at 14 24 32](https://cloud.githubusercontent.com/assets/1119271/11309538/70be96e0-8f92-11e5-9f25-eca72eefa61b.png)

In WP Admin, you can access the "Image Details" screen by clicking on the pencil icon in the image toolbar. ([1m36s](https://cloudup.com/cp5LXZDQrC6))

![screen shot 2015-11-20 at fri nov 20 2 16 58 pm](https://cloud.githubusercontent.com/assets/1119271/11309364/7a0888a6-8f91-11e5-8ffe-e3364f3c3095.png)

![screen shot 2015-11-20 at 13 58 47](https://cloud.githubusercontent.com/assets/1119271/11309369/816f50d4-8f91-11e5-9092-7b6bade9af0f.png)

Recently, the most-requested items have been:
- Link To 
- Image Border & Color
- Image Margins

They also expect to be able to change: 
- Caption
- Alt text

Other things are used and requested less frequently to my knowledge.

**Comment by mikejolley** [comment]
Also seen this request in ticket 982616. For now I'm recommending they switch to the old editor to access the link-to functionality.

**Comment by eliorivero** [comment]
User at 2437867-t complains about missing image editing features, in this case regarding cropping, flipping and rotation.

**Comment by carinapilar** [comment]
User missing the "Link To" feature <a href="https://en.forums.wordpress.com/topic/categories-and-tag-converter-issue?replies=13#post-2537098">here</a>.

**Comment by aduth** [comment]
As an FYI, we do support linking to the original images by clicking the Add Link button in the toolbar after selecting an image. Doing so will prepopulate the URL field with a link to the full-size image. This is a common use-case, though we do intend to add support for other "Link to" options (attachment page, custom URL, etc).

**Comment by carinapilar** [comment]
Yeah, the problem is that you have to do that in all images, add image then add the link. Before, if you set "Link To" to "Media File" in the first one, when adding other images it was loading the same option as default.

**Comment by jeherve** [comment]
User missing the "Link To" feature [here](https://wordpress.org/support/topic/centralized-posting-a-few-quetions).

**Comment by formosattic** [comment]
User in 2621414-t is missing "Link To" for media file. A feature he relies on a weekly basis.

**Comment by ryanboren** [comment]
I'm trying to improve the accessibility of my posts my adding alt to my images. I don't see how to do that in Calypso. I usually drag-and-drop images. The inline image toolbar allows adding a caption, but not alt.

**Comment by ryanboren** [comment]
I added accessibility because alt provisioning is not part of drag-and-drop image flow. Encourage accessible images.

**Comment by ryanboren** [comment]
Giving this a bump because of the accessibility implications. 

The lack of a modal means those who drag-and-drop images directly into the editor can't add alt text to dropped images without rummaging in the insert image modal (accessed via a + plus button).

Further, without access to the image details modal from the editor’s inline image toolbar, adding alt to images on old posts requires scrolling through the image modal looking for the right image amidst months of thumbnails. Unlike in core WP, there is no “Images uploaded to this post” filter. This is supremely frustrating flow if you trying to be good about alt text.

https://test.wordpress.com/2016/09/30/session-notes-drag-and-drop-image-flow-trackers-editor-masterbar-macnchrome-iphone-7/

**Comment by kriskarkoski** [comment]
+1 to adding this soon

Link To in paritcular is asked by users quite frequently and while it can be done in the Calypso Editor by clicking the image, then the link icon, having to do this for every image is tedious and it’s not intuitive to users how to do this, especially those used to the options already in the WP Admin editor.

**Comment by shaunandrews** [comment]
A quick idea for adding the details modal:

![image selection and details](https://cloud.githubusercontent.com/assets/191598/23678240/572f163a-0351-11e7-9a00-878fddff946b.gif)

**Comment by drw158** [comment]
Here is what I suggest for tackling this.

1. We need to at least add an edit button to the popover that is shown when you click an image in the editor. Missing that button is a broken experience.

![fullsizerender 6](https://cloud.githubusercontent.com/assets/618551/24017169/ef1be286-0a5c-11e7-8d1c-4263e5b109e7.jpg)


Next, it helps me to think of things that affect the original image, and things that are only relevant to this post.

2. I propose that we add a new tab to the image details modal that contains all of the post specific settings:

![fullsizerender 5](https://cloud.githubusercontent.com/assets/618551/24016992/4488eddc-0a5c-11e7-8038-f5b38d22afb6.jpg)

I'm not sure `Post settings` is the right name, but `Display settings` doesn't make much sense if it contains the link setting.

`Image settings` contains all of the existing info like caption and alt text.

We could also add in image size and image alignment, but this would just be the secondary way to edit these settings. The primary way would be to do it from the popover in the editor.

3. Next, I'd try to bring as many of the more often used settings into the editor, like Link to:

![fullsizerender 8](https://cloud.githubusercontent.com/assets/618551/24017158/e466c93c-0a5c-11e7-9795-6c2837b50188.jpg)

Clicking the link icon would bring up a variation of the link modal:

![fullsizerender 7](https://cloud.githubusercontent.com/assets/618551/24017295/6bce5af2-0a5d-11e7-93d8-426269b00bed.jpg)

This way it's more obvious how to link an image, and they can do it right from the editor.

**Comment by drw158** [comment]
Can this issue be filed under a Project, or can it be assigned to someone? It seems that Delta is already going to handle this issue, but I didn't know until reading through a P2.

**Comment by msmithgu** [comment]
This issue is partly covered by Issue https://github.com/Automattic/wp-calypso/issues/8306 and PR https://github.com/Automattic/wp-calypso/issues/13479 , which adds an inline image edit button, but does not provide a Link To feature. We may want to create a separate issue for a Link To button.

**Comment by jblz** [comment]
@msmithgu @gwwar -- is anything still outstanding here (or in #8306)?

Since #14143 got merged and we're tracking issues in (at least) #14233, are the above issues ok to close?

**Comment by gwwar** [comment]
@jblz, yes I think it makes sense to close these out and add new issues for additional improvements.

**Comment by jblz** [comment]
Sounds good. Thanks! If anyone disagrees, please feel free to reopen :)
